### PR TITLE
[WIP] connection not open on send?

### PR DESCRIPTION
**Pull request by jellegerbrandy**
- better error propagation of (some) connection problems

### CODE REVIEW

File: src/components/Shared/withSubscription.tsx [hunk 1]
```diff
@@ -110,11 +110,12 @@ const withSubscription = <Props extends ISubscriptionProps<ObservableType>, Obse
...
           // eslint-disable-next-line no-console
           console.error(getDisplayName(wrappedComponent), "Error in subscription", error);
-
-          this.setState({
-            isLoading: false,
-            error,
-          });
+          // this trick will effectively propagate the error to the error page (its magic!)
+          this.setState(() => { throw error; });
+          // this.setState({
```
Comment (dkent600) [inline src/components/Shared/withSubscription.tsx]: remove commented code?

File: src/layouts/Header.tsx [hunk 2]
```diff
@@ -84,6 +84,8 @@ class Header extends React.Component<IProps, IStateProps> {
     enableWalletProvider({
       suppressNotifyOnSuccess: true,
       showNotification: this.props.showNotification,
+    }).catch((err) => {
```
Comment (dkent600) [inline src/layouts/Header.tsx]: `enableWalletProvider` never throws an exception.  Instead it shows a notification and returns false (so we are already feeding back to the user)

File: src/App.tsx [hunk 3]
```diff
@@ -40,6 +40,7 @@ export class App extends React.Component<{}, {
       .catch ((err): void => {
```
Comment (dkent600) [inline src/App.tsx]: `initializeArc` doesn't throw exceptions.  It returns false if it fails.  Note the `while` loop.

**Comment by jellegerbrandy** [comment]
this PR is not relevant anymore - the behavior was improved in anoth PR #1213

**Pull request closed without merging**